### DISCUSSION TITLE
Throw exception from Twig

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#9001] [Ideas]
It would be useful if we could throw exceptions using Twig function - it would be helpful for reusable Twig components, to throw errors informing on incorrect or missing parameters passed to component.

**Comment by brandonkelly** [comment]
Are you aware that it’s possible to throw exceptions using the [{% exit %}](https://craftcms.com/docs/3.x/dev/tags.html#exit) tag?

**Comment by piotrpog** [comment]
@brandonkelly 
Can you set custom exception message using it though?

**Comment by brandonkelly** [comment]
@piotrpog No, which is why I left this opened. But just making sure you’re aware of that option. It will throw the appropriate HTTP exception class depending on the response code you pass in (e.g. 404 = NotFoundHttpException).

**Comment by piotrpog** [comment]
@brandonkelly 
Sure. But i thought that just throwing error 500 (i would imagine this one would be appropriate for missing variables) might be a little too vague in some cases.
